tests/unit/test_supermetrics.py

```diff
@@ -0,0 +1,171 @@
+    return Supermetrics(
+        credentials={
+            "user": supermetrics_credentials.user,
+            "api_key": supermetrics_credentials.api_key
+        },
+        query_params={"ds_id": "GA", "query": "test_query"}
+    )
+
+def test_to_json(mocker, supermetrics):
+    # Mock the handle_api_response function to simulate an API response
+    mock_handle_api_response = mocker.patch('viadot.sources.supermetrics.handle_api_response')
+    mock_response = {
+            "data": [["value1", "value2"]],
+            "meta": {
+                "query": {
+                    "fields": [
+                        {"field_name": "col1"},
+                        {"field_name": "col2"}
+                    ]
+                }
+            }
+        }
+    # Set the mock to return the mock response object
+    mock_handle_api_response.return_value.json.return_value = mock_response
+
+    # Call the method under test
+    response = supermetrics.to_json()
+
+    # Assert that the response is as expected
+    assert response == {
+        "data": [["value1", "value2"]],
+        "meta": {
+            "query": {
+                "fields": [
+                    {"field_name": "col1"},
+                    {"field_name": "col2"}
+                ]
+            }
+        }
+    }
+
+def test_to_df_with_data(supermetrics, mocker):
+    # Mock the API response with some data
+    mock_response = {
+                "meta": {
+                    "query": {
+                        "ds_id": "GA",  #Data source ID, e.g., Google Analytics
+                        "fields": [
+                            {
+                                "field_name": "date",
+                                "field_type": "DIMENSION",
+                                "field_split": "row"
+                            },
+                            {
+                                "field_name": "sessions",
+                                "field_type": "METRIC",
+                                "field_split": "row"
+                            }
+                        ],
+                        "other_query_metadata": "..."
+                    },
+                    "status": "success",  # Status of the query
+                    "execution_time": "0.456"  # Time taken to execute the query
+                },
+                "data": [
+                    ["2023-01-01", 100],  # Example data rows
+                    ["2023-01-02", 150],
+                    ["2023-01-03", 120]
+                ],
+                "paging": {
+                    "current_page": 1,  # Current page number if pagination is used
+                    "total_pages": 1,
+                    "total_results": 3
+                }
+            }
+
+    mock_method = mocker.patch('viadot.sources.supermetrics.Supermetrics.to_json')
+    mock_method.return_value = mock_response
+    mock_method = mocker.patch('viadot.sources.supermetrics.Supermetrics._get_col_names')
```

Comment:
See the comment above about 2 API calls; this is related - this shouldn't have to be mocked as I would assume such method doesn't need to perform an API call - it should accept a param such as `response: dict[str, Any]` and extract the column names from it.

Reply:
Created an issue to solve this https://github.com/dyvenia/viadot/issues/1023